chasm/lib/activity/frontend.go

```diff
@@ -22,26 +31,142 @@ type FrontendHandler interface {
+// It validates the request, resolves the namespace ID, applies default configurations,
+// and forwards the request to the activity service client.
+//
+// The method performs the following steps:
+// 1. Resolves the namespace name to its internal ID
+// 2. Validates and populates request fields (timeouts, retry policies, search attributes). The request is cloned
+// before mutation to preserve the original for retries.
+// 3. Sends the request to the history activity service.
+//
+// Parameters:
+//   - ctx: Context for the request, used for cancellation and deadlines
+//   - req: StartActivityExecutionRequest containing activity details, options, and metadata
+//
+// Returns:
+//   - *workflowservice.StartActivityExecutionResponse: Response containing activity execution details
+//   - error: Any error encountered during namespace resolution, validation, or execution
 func (h *frontendHandler) StartActivityExecution(ctx context.Context, req *workflowservice.StartActivityExecutionRequest) (*workflowservice.StartActivityExecutionResponse, error) {
 	namespaceID, err := h.namespaceRegistry.GetNamespaceID(namespace.Name(req.GetNamespace()))
 	if err != nil {
 		return nil, err
 	}
 
+	modifiedReq, err := h.validateAndPopulateListRequest(req, namespaceID)
+	if err != nil {
+		return nil, err
+	}
+
 	resp, err := h.client.StartActivityExecution(ctx, &activitypb.StartActivityExecutionRequest{
 		NamespaceId:     namespaceID.String(),
-		FrontendRequest: req,
+		FrontendRequest: modifiedReq,
 	})
 
 	return resp.GetFrontendResponse(), err
 }
+
+func (h *frontendHandler) validateAndPopulateListRequest(
+	req *workflowservice.StartActivityExecutionRequest, namespaceID namespace.ID) (*workflowservice.StartActivityExecutionRequest, error) {
+	// Since validation includes mutation of the request, we clone it first so that any retries use the original request.
+	req = common.CloneProto(req)
+
+	activityType := ""
+	if req.ActivityType != nil {
+		activityType = req.ActivityType.GetName()
+	}
+
+	if req.Options.RetryPolicy == nil {
+		req.Options.RetryPolicy = &commonpb.RetryPolicy{}
+	}
+
+	modifiedAttributes, err := ValidateActivityRequestAttributes(
+		req.ActivityId,
+		activityType,
+		dynamicconfig.DefaultActivityRetryPolicy.Get(h.dc),
+		dynamicconfig.MaxIDLengthLimit.Get(h.dc)(),
+		namespaceID,
+		req.Options,
+		req.Priority,
+		durationpb.New(0),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the command attributes with the adjusted timeouts
+	if modifiedAttributes != nil {
+		req.Options.ScheduleToCloseTimeout = modifiedAttributes.ScheduleToCloseTimeout
+		req.Options.ScheduleToStartTimeout = modifiedAttributes.ScheduleToStartTimeout
+		req.Options.StartToCloseTimeout = modifiedAttributes.StartToCloseTimeout
+		req.Options.HeartbeatTimeout = modifiedAttributes.HeartbeatTimeout
+	}
+
+	modifiedSaAttributes, err := ValidateStandaloneActivity(
```

Comment:
You're using "SA" here to mean Standaloe Activity, but earlier in the PR and elsewhere in the codebase it means "Search Attribute".